### PR TITLE
Update project

### PR DESCRIPTION
**Pull request by nisrulz**
This PR updates this project:
- Kotlin v2.0.0
- Compile SDK to 34
- AGP to v8.5.0
- Compose Gradle Plugin v2.0.0
- Gradle wrapper v8.8
- Upgrade all other dependencies
- Replace Accompanist with Compose Foundation APIs
- Upgrade Exoplayer APIs to latest version
- Remove more deprecated code
- Added editorconfig file to handle Composable functions in IDE when using KtLint Plugin
- Cleanup build.gradle files, extract common config into common-config.gradle file

### CODE REVIEW

File: app/src/main/java/com/smarttoolfactory/composebeforeafter/ContentScaleSelectionMenu.kt [hunk 1]
```diff
@@ -72,7 +82,10 @@ fun ExposedSelectionMenu(
         fontSize = 14.sp
     ),
     colors: TextFieldColors = ExposedDropdownMenuDefaults.textFieldColors(
-        containerColor = Color.Transparent,
```
Comment (nisrulz) [inline app/src/main/java/com/smarttoolfactory/composebeforeafter/ContentScaleSelectionMenu.kt:75]: This property doesn't exists anymore.

File: app/src/main/java/com/smarttoolfactory/composebeforeafter/ui/theme/Theme.kt [hunk 2]
```diff
@@ -49,14 +49,20 @@ fun ComposeBeforeAfterTheme(
...
     }
     val view = LocalView.current
     if (!view.isInEditMode) {
+        val currentWindow =
+            (view.context as? Activity)?.window
+                ?: throw Exception("Not in an activity - unable to get Window reference")
+
         SideEffect {
-            (view.context as Activity).window.statusBarColor = colorScheme.primary.toArgb()
-            ViewCompat.getWindowInsetsController(view)?.isAppearanceLightStatusBars = darkTheme
```
Comment (nisrulz) [inline app/src/main/java/com/smarttoolfactory/composebeforeafter/ui/theme/Theme.kt:59]: Deprecated code.

File: beforeafter/src/main/java/com/smarttoolfactory/beforeafter/ZoomState.kt [hunk 3]
```diff
@@ -154,17 +153,16 @@ fun rememberZoomState(
  * @param rotationEnabled when set to true rotation is enabled
  */
 @Immutable
-class ZoomState internal constructor(
+open class ZoomState internal constructor(
```
Comment (nisrulz) [inline beforeafter/src/main/java/com/smarttoolfactory/beforeafter/ZoomState.kt:156]: Needed, because:

Screenshot 2024-06-18 at 18 40 23

**Comment by nisrulz** [comment]
Regarding this:

Screenshot 2024-06-18 at 18 47 08

Open Question: 
I switched to latest Exoplayer which is part of [Media3 libraries](https://github.com/androidx/media). 

Without knowing what the bug was, I cannot declare it is fixed in the current version. Readme in this project doesn't link to the right question anymore and the link is broken. This is why I have not removed it from Readme yet.

If the original author of the lib can verify the bug is fixed, I can also remove this info from  Readme in this PR :)

**Comment by SmartToolFactory** [comment]
Thank you very much for your contributions.

**Pull request merged**